src/main/java/org/sonarlint/intellij/telemetry/SonarLintTelemetry.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.telemetry;
+
+import com.intellij.concurrency.JobScheduler;
+import com.intellij.openapi.application.PathManager;
+import com.intellij.openapi.components.ApplicationComponent;
+import com.intellij.openapi.project.ProjectManager;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.NotNull;
+import org.sonarlint.intellij.SonarApplication;
+import org.sonarsource.sonarlint.core.WsHelperImpl;
+
+public class SonarLintTelemetry implements ApplicationComponent {
+  private static final String STORAGE_FILENAME = "sonarlint_usage.json";
+  private final TelemetryClient worker;
+  private TelemetryStorage storage;
+  private ScheduledFuture<?> scheduledFuture;
+  private boolean error = false;
+
+  public SonarLintTelemetry(SonarApplication application, ProjectManager projectManager) {
+    this.worker = new TelemetryClient(new WsHelperImpl(), application, projectManager);
+  }
+
+  private void loadStorage() {
+    Path filePath = getStorageFilePath();
+
+    if (Files.exists(filePath)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)